apps/cowswap-frontend/src/modules/fiatAmount/hooks/useFiatAmount.ts

```diff
@@ -0,0 +1,32 @@
+import { useMemo } from 'react'
+
+import { Currency, CurrencyAmount, Token } from '@uniswap/sdk-core'
+
+import { Nullish } from 'types'
+
+import { currencyAmountToTokenAmount } from 'utils/currencyAmountToTokenAmount'
+
+import { useUsdcPrice } from './useUsdcPrice'
+
+export interface FiatAmountInfo {
+  value: CurrencyAmount<Token> | null
+  isLoading: boolean
+}
+
+const defaultFiatAmountState = { value: null, isLoading: false }
+
+export function useFiatAmount(_amount: Nullish<CurrencyAmount<Currency>>): FiatAmountInfo {
+  const amount = currencyAmountToTokenAmount(_amount)
+  const usdcPrice = useUsdcPrice(amount?.currency)
```

Comment:
The use of the terms fiat amount and usdc amount are a bit confusing to me.
It starts off with fiat amounts, then converted to usdc amounts and finally here is fiat amount again.
Is that expected?

Reply:
https://cowservices.slack.com/archives/C0361CDG8GP/p1693293238882729